### ISSUE TITLE
Not working in Firefox

### ISSUE THREAD

**Issue by mlukasch** [react-dnd/react-dnd#869]
Hello,
reactdnd is not working anymore in the latest Firefox-Version 55.0.3
Using reactdnd version:
 "react-dnd": "^2.5.1",
 "react-dnd-html5-backend": "^2.5.1",

Does anyone knows the latest version of reactdnd where Firefox is working?

**Comment by mlukasch** [comment]
React-dnd does work however in Firefox 55.0.2.

**Comment by AKarmanov** [comment]
Doesn't work in 56.0.0 and beta 57.0.0 for me

**Comment by Paddy-Hamilton** [comment]
not working in 56.0 either

**Comment by alexfi** [comment]
Any ideas on this one?

**Comment by bjrmatos** [comment]
can someone point to a repository with a sample case in which react-dnd is failing? i have an app that uses react-dnd a lot but i don't see this problem in firefox 56. 

also please make sure to specify the OS that you are using, i have seen that recently html5 Drag and Drop API is having some inconsistencies in different OS

**Comment by lvpro** [comment]
I've observed issues with FF 56 on Linux, as well (didn't test on Windows or OSX).

http://react-dnd.github.io/react-dnd/examples-sortable-simple.html

Trying that simple example, an element will move at max 1 position and then no longer shift elements.

**Comment by erwinverdonk** [comment]
For our case - DnD not working in FF, example https://react-dnd.github.io/react-dnd/examples-drag-around-naive.html - I did fix the issue with the pull request I submitted yesterday. Perhaps you could try and report back whether it solves your case as well.

**Comment by bozskejkaja** [comment]
Still having issues on FF 57.0.1 on Ubuntu 16.04. I'm not able to drag & drop at all, seems like the drag event starts but cancels immediately afterwards. Works fine in chrome.

**Comment by yankopetrov** [comment]
I also have problems with  FF and OSX 10.13.1

**Comment by Paddy-Hamilton** [comment]
@yankopetrov 

I'm still experiencing the problems in Nightly, but it seems to break a whole lot more in the app than just RDandD. So not sure I trust that as an indication if FF is the real cause of the bug, and if they have fixed it.

**Comment by Paddy-Hamilton** [comment]
@erwinverdonk , could you share your fix ? what was the PR?
cheers

**Comment by yankopetrov** [comment]
@Paddy-Hamilton yes with FF Nightly still some bugs appears. Working with chrome and safari like a charm so should be FF issue. My case is exact the one in the reported FF issue.

**Comment by erwinverdonk** [comment]
@Paddy-Hamilton: Yeah I see it did not auto link here, so here is the link to PR:  https://github.com/react-dnd/react-dnd/pull/928

**Comment by dstril** [comment]
So does anyone know what is the issue?
I'm using FF 57.0.4 64bit on Ubuntu 17.04 and it's still not working. Even this example: [http://react-dnd.github.io/react-dnd/examples-chessboard-tutorial-app.html](url)

**Comment by dmitrystril** [comment]
Hi, I've found some sort of workaround that worked for me. My drag-n-drop worked fine in Chrome but didn't work in FF at all.
So my code before: 
`render() { 
return ( isDragging ? null :
                <div className='myDraggableComponent'>...</div>)}`
And after:
`render() { 
return <div className='myDraggableComponent' style={{ visibility: isDragging ? 'hidden' : 'visible' }}>...</div>)}`

**Comment by Paddy-Hamilton** [comment]
@dmitrystril @yankopetrov 

The issue was still their for me in FF Nightly so I had to look for alternatives, I came across [react-beautiful-dnd](https://github.com/atlassian/react-beautiful-dnd) and it works like a charm in FF as well as all other major browsers, and has less markup than react-dnd which is a bonus.

**Comment by adamhooper** [comment]
The chess tutorial does not work on Linux with FF 60.0:
1. Click and hold the knight
2. Move to a valid square (one that renders green)
3. Move to an adjacent square

Expected results: the square is highlighted red
Actual results: the green square stays highlighted, forever

Also: no matter what square my cursor is on, releasing the mouse cancels the drag.

**Comment by darthtrevino** [comment]
Ae you on Linux @adamhooper? I can't reproduce in OSX

**Comment by adamhooper** [comment]
Fedora 28

**Comment by darthtrevino** [comment]
I'm going to look into different cross-browser testing tools we can use to catch issues like this. The existing examples should be a pretty good baseline for tests.

**Comment by lvpro** [comment]
I can confirm I've only seen this problem on the Linux version of Firefox.  Works fine on Windows.

**Comment by koox00** [comment]
If you look at the `sortable` examples `simple` doesn't work, but `stress test` works.
The difference is that `stress test` is updating the state inside a `requestAnimationFrame`.

Using `requestAnimationFrame` solves it for me.
Firefox on Linux.

**Comment by koox00** [comment]
@darthtrevino yeah of course!

**Comment by Skylsmoi** [comment]
I just ran into a similar issue.
Although I already had latest versions of Firefox and react dnd (on linux).

What was causing my issue was that my drag handler (component with `ref={props.connectDragSource}`) was inside of a `<Link>` from react router.

I don't know why it was causing the issue but as soon as I put it outside the  `<Link>`, everything worked fine.

**Comment by SystemParadox** [comment]
I've got the same issue in Firefox 60.8.0esr on Debian 10.

Updating the state inside `requestAnimationFrame` fixes it for Firefox, but it makes dragging and Chrome sluggish and broken.

**Comment by SystemParadox** [comment]
Please do not close this. It is still broken.

**Comment by darthtrevino** [comment]
@SystemParadox What OS are you using?

**Comment by SystemParadox** [comment]
Linux (Debian 10). I haven't had chance to try it on Windows yet but it seem very strange for it to be different.

**Comment by Spaubleit** [comment]
All works fine in Firefox 69.0.1
Broken in Firefox Nightly 70.0a1 (2019-08-21)

**Comment by SystemParadox** [comment]
- Works in 68.10.0esr
- Works in 78.0.2

If they broke it in 70.0a1 then it would seem this keeps breaking for some reason, which is a bit concerning. If it breaks again then we really need to create a bare minimum test case and submit a ticket to the firefox devs.

**Comment by devyeshtandon** [comment]
It's 2021 and it still does not work. 
Firefox version: 92.0
OS: Ubuntu 18.04
Link: https://codesandbox.io/s/github/react-dnd/react-dnd/tree/gh-pages/examples_hooks_js/00-chessboard?from-embed

**Comment by darthtrevino** [comment]
I haven't had time to dig into a triage here, although I can confirm this is a case in Windows as well with FF 92. PRs are welcome.

**Comment by markbeacom** [comment]
For anyone who is still experiencing this issue, i found that on firefox, using a button with the ref drag will not work sadly.

Not exactly sure why this is the case but i avoided the issue by just replacing the button with a div and remove  the button element.

**Comment by LeeeeeeM** [comment]
> For anyone who is still experiencing this issue, i found that on firefox, using a button with the ref drag will not work sadly.
> 
> Not exactly sure why this is the case but i avoided the issue by just replacing the button with a div and remove the button element.

maybe caused by this  https://github.com/w3c/pointerevents/issues/177  https://github.com/whatwg/html/issues/2368

**Comment by LeeeeeeM** [comment]
can use css  ```point-event: none;``` to solve this firefox bug

**Comment by Yohandah** [comment]
I have the same issue and using a div instead of a button seems to do the trick even tho I would like to use a button.

**Comment by Yohandah** [comment]
> can use css `pointer-event: none;` to solve this firefox bug

Can you elaborate please ? You put this CSS Prop on the `<button>` receiving the drop ref ? `<button ref={dropRef} style={{ pointerEvent: 'none'}}>` ?

**Comment by msdrigg** [comment]
2025 and still seeing issues with firefox. Chess example works, but my own code does not work unfortunately and it's only ff that's the issue.

**Comment by justandras** [comment]
What I found is that in Chrome I can drag the elements by clicking anywhere, in Firefox however only text elements drag in my case.

![Image](https://github.com/user-attachments/assets/b38c647a-efcd-4a5e-991f-f44780307422)
